sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchTcpImpl.java

```diff
@@ -0,0 +1,652 @@
+    return promise.future();
+  }
+
+  /**
+   * This method will do the bulk data insertion.
+   *
+   * @param index String index name
+   * @param dataList List<Map<String, Object>>
+   * @return boolean
+   */
+  @Override
+  public Future<Boolean> bulkInsert(String index, List<Map<String, Object>> dataList) {
+    Promise<Boolean> promise = Futures.promise();
+    long startTime = System.currentTimeMillis();
+    ProjectLogger.log(
+        "ElasticSearchTcpImpl:bulkInsert: method started at ==" + startTime + " for index " + index,
+        LoggerEnum.PERF_LOG);
+    promise.success(true);
+    try {
+      BulkProcessor bulkProcessor =
+          BulkProcessor.builder(
+                  ConnectionManager.getClient(),
+                  new BulkProcessor.Listener() {
+                    @Override
+                    public void beforeBulk(long executionId, BulkRequest request) {}
+
+                    @Override
+                    public void afterBulk(
+                        long executionId, BulkRequest request, BulkResponse response) {
+                      Iterator<BulkItemResponse> bulkResponse = response.iterator();
+                      if (bulkResponse != null) {
+                        while (bulkResponse.hasNext()) {
+                          BulkItemResponse bResponse = bulkResponse.next();
+                          ProjectLogger.log(
+                              "ElasticSearchTcpImpl:bulkInsert: "
+                                  + "Bulk insert api response==="
+                                  + bResponse.getId()
+                                  + " "
+                                  + bResponse.isFailed());
+                        }
+                      }
+                    }
+
+                    @Override
+                    public void afterBulk(
+                        long executionId, BulkRequest request, Throwable failure) {
+                      ProjectLogger.log(
+                          "ElasticSearchTcpImpl:bulkInsert: Bulk upload error block", failure);
+                    }
+                  })
+              .setBulkActions(10000)
+              .setConcurrentRequests(0)
+              .build();
+
+      for (Map<String, Object> map : dataList) {
+        map.put(JsonKey.IDENTIFIER, map.get(JsonKey.ID));
+        IndexRequest request =
+            new IndexRequest(index, _DOC, (String) map.get(JsonKey.IDENTIFIER)).source(map);
+        bulkProcessor.add(request);
+      }
+      // Flush any remaining requests
+      bulkProcessor.flush();
+
+      // Or close the bulkProcessor if you don't need it anymore
+      bulkProcessor.close();
+
+      // Refresh your indices
+      ConnectionManager.getClient().admin().indices().prepareRefresh().get();
+    } catch (Exception e) {
+      promise.failure(e);
+      promise.success(false);
+      ProjectLogger.log(e.getMessage(), e);
+    }
+    long elapsedTime = ElasticSearchHelper.calculateEndTime(startTime);
```

Comment:
remove above line and directly use method inside logger.

Reply:
Done